lib/inventory_refresh/application_record_iterator.rb

```diff
@@ -17,19 +25,28 @@ def initialize(inventory_collection: nil)
     # and relation.where(:id => 500ids)
     #
     # @param batch_size [Integer] A batch size we want to fetch from DB
-    # @param attributes_index [Hash{String => Hash}] Indexed hash with data we will be saving
     # @yield Code processing the batches
-    def find_in_batches(batch_size: 1000, attributes_index: {})
-      attributes_index.each_slice(batch_size) do |batch|
-        yield(inventory_collection.db_collection_for_comparison_for(batch))
+    def find_in_batches(batch_size: 1000)
+      if iterator
+        iterator.call do |batch|
+          yield(batch)
+        end
+      elsif query
+        manager_uuids_set.each_slice(batch_size) do |batch|
+          yield(query.where(inventory_collection.targeted_selection_for(batch)))
+        end
+      else
+        manager_uuids_set.each_slice(batch_size) do |batch|
+          yield(inventory_collection.db_collection_for_comparison_for(batch))
+        end
```

Comment:
Are these batch saving changes from a specific commit or series of commits that we can revert instead?  It's hard to tell if this a revert or new code.

Reply:
Not new code, I can try reverting every PR between 0.3.z and 0.2.z instead

Reply:
Okay done, there were ~20 PRs that had to be reverted and I had to cherry-pick NickL's rails 6 support PR which went straight to 0.2.z for some reason

Reply:
I'm open to whatever path is easier to track and do.  It was more a question as it looked like reverts of feature commits.

Reply:
Yeah reverting each is definitely easier to see what happened